pkg/clustermanager/reconcile_clusterprovider.go

```diff
@@ -131,6 +134,7 @@ func (c *controller) handleDelete(providerName string) error {
 		}
 	}
 	if isNsEmpty {
+		delete(c.providers, providerName)
```

Comment:
We need to delete from cache unconditionally, otherwise we may have cached provider which object is gone. Also we should delete under lock.